### ISSUE TITLE
Disabling embed blocks breaks pasting links in RichText

### ISSUE THREAD

**Issue by walbo** [WordPress/gutenberg#27835]
When embed blocks are disabled and you try to paste a text that has a link on a seperate line, nothing is pasted.

**To reproduce**
1. Disable embed blocks. (to make it easy to reproduce I only allow the paragraph block)
    ```php
    add_filter( 'allowed_block_types', function() {
       return [ 'core/paragraph' ];
    } );
    ```
1. Create a new page and try to paste the text:
    ```
    Lorem ipsum dolor sit amet.

    https://wordpress.org/news/2020/12/state-of-the-word-2020/

    Lorem ipsum dolor sit amet.
    ```
    *(it also don't work to just paste the link)*

**Expected behavior**
Text and a normal link should be inserted into the editor.

**Editor version:**
Tested both in WP 5.5/5.6 and with Gutenberg 9.5.2

**Additional info:**
From the console:
```
Received HTML:

```
```
Received plain text:

Lorem ipsum dolor sit amet.

https://wordpress.org/news/2020/12/state-of-the-word-2020/

Lorem ipsum dolor sit amet.
```
```
Processed HTML piece:

<p>Lorem ipsum dolor sit amet.</p><p>https://wordpress.org/news/2020/12/state-of-the-word-2020/</p><p>Lorem ipsum dolor sit amet.</p>
```

**Comment by Mamaduka** [comment]
Related #11245.

**Comment by kathrynwp** [comment]
I'm still able to replicate with:

WordPress 6.0.1
Gutenberg 13.8.1 active or inactive
No other plugins active

Two notes:

1) I **am** able to paste in the text on its own, but not with the link

2) There's a deprecation warning caused by the test function above:

`Deprecated: Hook allowed_block_types is deprecated since version 5.8.0! Use allowed_block_types_all instead. in /Users/kpmba/Sites/WordPressDev9/wp-includes/functions.php on line 5754`

Changing the first line to the following gets rid of the warning:

`add_filter( 'allowed_block_types_all', function() {`

**Comment by richardmorrison** [comment]
I am also still able to replicate this with:

WordPress 6.0.1
Gutenberg 13.9.0
Only Hello Dolly active

Disabled the Embed blocks via the vertical ... menu for Gutenberg on page editor, and Preferences, then unchecked the blocks.

<img width="1221" alt="Screen Shot 2022-08-17 at 4 37 11 PM" src="https://user-images.githubusercontent.com/61720320/185035905-1933c939-af54-4dce-9b66-47c60a7b5b04.png">

I can paste the content with link form initial issue message in a new paragraph block, and nothing appears.

**Console message:**

Received plain text:

 Lorem ipsum dolor sit amet.

https://wordpress.org/news/2020/12/state-of-the-word-2020/

Lorem ipsum dolor sit amet.
blocks.min.js?ver=658a51e7220626e26a92a46af5c2e489:3 Processed HTML piece:

 <p>Lorem ipsum dolor sit amet.</p><p>https://wordpress.org/news/2020/12/state-of-the-word-2020/</p><p>Lorem ipsum dolor sit amet.</p>

**Follow up**

Re-enabling the embed blocks, the link correctly pastes, and even loads a preview of the link, when pasted into a plain paragraph block.

**Comment by ghm-jpeterman** [comment]
I'm able to reproduce this as well using Wordpress 6.0.1.

The core/embed plugin is disabled using the `allowed_block_types_all` filter. The following blocks are active:
```
'core/paragraph',
'core/heading',
'core/list',
'core/quote',
'core/table'
```
**I can copy and paste an http link** 
```
http://wordpress.org
```

**I cannot copy and paste an https link**
```
https://wordpress.org
```

**I cannot copy and paste text that has a link on it's own line**
```
This is text.

https://wordpress.org

This is text.
```

**I can copy and paste text with a link on it's own line if I add a character before it**
```
This is text.

a https://wordpress.org

This is text.
```

If I reenable the `core/embed` plugin, all of the above examples work.

**Comment by annezazu** [comment]
@ellatrix I see quite a while ago you self assigned this, are you still able to take a look, considering it seems to still be an active problem? I imagine this is a pretty complex one to solve considering how the embed block is setup.

**Comment by ghm-jpeterman** [comment]
@annezazu Thanks for your response. Is there anyone else that may be able to pick this up since we haven't heard back from @ellatrix?

**Comment by annezazu** [comment]
Hi @ghm-jpeterman! There are a lot of priorities in place right now ahead of feature freeze for 6.1 so I'll flag this in the Core Editor slack channel for now but keep in mind that this may not be a priority currently.

**Comment by ghm-jpeterman** [comment]
Hey @annezazu! Totally understand. Thanks for flagging!

**Comment by Mamaduka** [comment]
@annezazu, can we add this to the 6.1 board? I might have a possible solution and will try to push PR this week.

**Comment by annezazu** [comment]
Fantastic. Adding now. Thank you @Mamaduka. @ghm-jpeterman - it would be great to help test the PR George comes up with when the time comes to help move the fix forward.

**Comment by ghm-jpeterman** [comment]
@annezazu, Sounds good. Thank you @Mamaduka!

**Comment by Mamaduka** [comment]
PR is ready for testing #43930.

**Comment by cbravobernal** [comment]
Bumped to 6.2 as mentioned in the PR https://github.com/WordPress/gutenberg/pull/43930.

**Comment by talldan** [comment]
I think there's some technical feedback in the closed PR -  [RichText: Check if the embed block can be inserted](https://github.com/WordPress/gutenberg/pull/43930#top).

tldr - the `htmlToBlocks` function that does the conversion checks for the existence of a matching `raw` transform that a block can register:
https://github.com/WordPress/gutenberg/blob/4af79288d4033198de603e255a9075ce82bded2c/packages/blocks/src/api/raw-handling/html-to-blocks.js#L32

In this case the embed block adds a raw transform, so it's present in the return value. The problem is that the list of transforms are not filtered by the which blocks are actually allowed. So the code creates an embed block regardless, and when insertion of that block is attempted, it fails.

Possibly two ways I can think to handle this.

First option might be to filter the list of available transforms by which blocks are allowed, so that the embed raw transform can never match if it's not allowed. The transforms would have to be filtered first in the block-editor, and then passed into the `pasteHandler`.

Another option might be to detect if the insertion of the embed block failed, and if so fallback to the original content.
